log4j-core/src/main/java/org/apache/logging/log4j/core/layout/StringBuilderEncoder.java

```diff
@@ -32,9 +32,14 @@
 public class StringBuilderEncoder implements Encoder<StringBuilder> {
 
     private static final int DEFAULT_BYTE_BUFFER_SIZE = 8 * 1024;
-    private final ThreadLocal<CharBuffer> charBufferThreadLocal = new ThreadLocal<>();
-    private final ThreadLocal<ByteBuffer> byteBufferThreadLocal = new ThreadLocal<>();
-    private final ThreadLocal<CharsetEncoder> charsetEncoderThreadLocal = new ThreadLocal<>();
+    /**
+     * This ThreadLocal uses raw and inconvenient Object[] to store three heterogeneous objects (CharEncoder, CharBuffer
```

Comment:
@jvz I can. Using just one is an optimization - ThreadLocalMap is polluted less, ThreadLocalMap.get() is called only once instead of three times.